Update WebGL 2.0.0 conformance test expectations due to new attribute drawingBufferColorSpace

squelart opened the pull request:
#### c9dd43753957fe16740e24b79074542865a2ae4f
<pre>
Update WebGL 2.0.0 conformance test expectations due to new attribute drawingBufferColorSpace
<a href="https://bugs.webkit.org/show_bug.cgi?id=248083">https://bugs.webkit.org/show_bug.cgi?id=248083</a>
rdar://102514357

Reviewed by Kimmo Kinnunen.

Monterey introduces new WebGL attributes (like WebGLRenderingContextBase.drawingBufferColorSpace) that the 2.0.0 conformance test context/constants-and-properties-2.html doesn&apos;t accept.
The corresponding 2.0.y test is already enabled, covers the same ground and accepts the new attributes.

* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/256930@main">https://commits.webkit.org/256930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/acc37ba78c644b3b426b0b569b591805f80d3633

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97273 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30419 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106791 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167059 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6830 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35271 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89666 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103474 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102936 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5132 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83899 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32114 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86983 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88804 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/554 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20277 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/537 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21734 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4778 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5337 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1781 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41059 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1762,7 +1762,9 @@ imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html [ Fa
-webkit.org/b/248083 [ BigSur+ ] webgl/2.0.0/conformance2/context/constants-and-properties-2.html [ Failure ]
+# Monterey introduces new WebGL attributes that 2.0.0 conformance tests don't accept. Corresponding 2.0.y tests cover the same ground and accept the new attributes.
+webkit.org/b/248083 [ Monterey+ ] webgl/2.0.0/conformance2/context/constants-and-properties-2.html [ Failure ]
+webkit.org/b/248083 [ Monterey+ ] webgl/2.0.y/conformance2/context/constants-and-properties-2.html [ Pass ]
```
This one is explicitly enabled in LayoutTests/TestExpectations.  Unless there's nothing extra that I'm missing, that should be enough as is, so this one is redundant.

kkinnunen-apple commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1762,7 +1762,9 @@ imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html [ Fa
 
-webkit.org/b/248083 [ BigSur+ ] webgl/2.0.0/conformance2/context/constants-and-properties-2.html [ Failure ]
+# Monterey introduces new WebGL attributes that 2.0.0 conformance tests don't accept. Corresponding 2.0.y tests cover the same ground and accept the new attributes.
+webkit.org/b/248083 [ Monterey+ ] webgl/2.0.0/conformance2/context/constants-and-properties-2.html [ Failure ]
```
The test webgl/2.0.0/conformance2/context/constants-and-properties-2.html is now incorrect, and does not need to be run.
So you can change LayoutTests/TestExpectations and add :

`webgl/2.0.0/conformance2/context/constants-and-properties-2.html [ Skip ]`

In section which says:
```
# WebGL 1.0.3 and 2.0.0 tests where behavior is obsolete and WebKit contains implementation
# and tests for the new behavior. Should be removed once 1.0.3 and 2.0.0 are retired.
```

squelart committed:
Update WebGL 2.0.0 conformance test expectations due to new attribute drawingBufferColorSpace
https://bugs.webkit.org/show_bug.cgi?id=248083
rdar://102514357

Reviewed by Kimmo Kinnunen.

Monterey introduces new WebGL attributes (like WebGLRenderingContextBase.drawingBufferColorSpace) that the 2.0.0 conformance test context/constants-and-properties-2.html doesn't accept.
The corresponding 2.0.y test is already enabled, covers the same ground and accepts the new attributes.

* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/256930@main
